diffusion/models/models.py

```diff
@@ -272,9 +273,38 @@ def stable_diffusion_xl(
         if isinstance(metric, CLIPScore):
             metric.requires_grad_(False)
 
+    # Convert string to list    
+    if isinstance(tokenizer_names, str):
+        tokenizer_names = [tokenizer_names]
+    
+    # Instantiate tokenizers
+    tokenizers = []
+    for tokenizer_name in tokenizer_names:
+            # If tokenizer_name_or_path contains more than one '/', then the string includes a subfolder to extract
+            path_split = tokenizer_name.split('/')
+            name = '/'.join(path_split[:2])
+            subfolder = '/'.join(path_split[2:]) if len(path_split) > 2 else ''
+
+            tokenizers.append(AutoTokenizer.from_pretrained(name, subfolder=subfolder))
+
     # Make the text encoder
-    tokenizer = SDXLTokenizer(model_name)
-    text_encoder = SDXLTextEncoder(model_name, encode_latents_in_fp16)
+    if isinstance(text_encoder_names, str):
+        text_encoder_names = [text_encoder_names]
+
+    torch_dtype = torch.float16 if encode_latents_in_fp16 else None
+    text_encoders = torch.nn.ModuleList()
+    for text_encoder_name in text_encoder_names:
+        # If tokenizer_name_or_path contains more than one '/', then the string includes a subfolder to extract
+        path_split = text_encoder_name.split('/')
+        name = '/'.join(path_split[:2])
+        subfolder = '/'.join(path_split[2:]) if len(path_split) > 2 else ''
+
+        # I'm not sure how to load this CLIP Text model using Auto...
```

Comment:
one way to achieve this (not sure if it is the cleanest way, so take it or leave it):

```
    torch_dtype = torch.float16 if encode_latents_in_fp16 else None
    text_encoders = torch.nn.ModuleList()
    d_text_encoder = 0
    d_model_keys = ['projection_dim', 'd_model', 'hidden_size'] # CLIP, T5, E5
    for text_encoder_name in text_encoder_names:
        # If tokenizer_name_or_path contains more than one '/', then the string includes a subfolder to extract
        path_split = text_encoder_name.split('/')
        name = '/'.join(path_split[:2])
        subfolder = '/'.join(path_split[2:]) if len(path_split) > 2 else ''

        # Get architecture and d_model for text encoder(s)
        text_encoder_config = PretrainedConfig.get_config_dict(name, subfolder=subfolder)[0] 
        
        d_model = None
        for d_model_key in d_model_keys:
            if d_model_key in text_encoder_config:
                # Add current text encoder d_model to total d_text_encoder
                d_text_encoder += text_encoder_config[d_model_key]
                break
        if not d_model:
            raise ValueError('Could not find any of d_model_keys in text encoder config')

        architectures = text_encoder_config['architectures']
        if architectures == ['CLIPTextModel']:
            text_encoders.append(CLIPTextModel.from_pretrained(name, subfolder=subfolder, torch_dtype=torch_dtype))
        elif architectures == ['CLIPTextModelWithProjection']:
            text_encoders.append(CLIPTextModelWithProjection.from_pretrained(name, subfolder=subfolder, torch_dtype=torch_dtype))
        else:
            text_encoders.append(AutoModel.from_pretrained(name, subfolder=subfolder, torch_dtype=torch_dtype))
```

this also loads each text encoder's config and tries to find the key that corresponds to the model output dim - will need this later to set the `cross_attention_dim` for the unet